### PR TITLE
Network interface documentation enhancements

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

File: doc/reference/networking/net_if.rst [hunk 1]
```diff
@@ -6,6 +6,50 @@ Network Interface
 Overview
 ********
 
+The network interface is a nexus that ties the network device drivers
+and the upper part of the network stack together. All the sent and received
+data is transferred via a network interface. The network interfaces cannot be
+created runtime as a special linker section will contain information about them
```
Comment (tbursztyka) [inline doc/reference/networking/net_if.rst]: at runtime?

Comment (dbkinder) [reply doc/reference/networking/net_if.rst]:
```suggestion
created at runtime.  A special linker section will contain information about them
```

File: doc/reference/networking/net_if.rst [hunk 2]
```diff
@@ -6,6 +6,50 @@ Network Interface
...
+:option:`CONFIG_NET_VLAN` is enabled. These macros are typically used in
+network device driver source code.
+
+The network interface can be turned ON by calling ``net_if_up()`` and OFF
+by calling ``net_if_down()``. When the device is powered ON, the network
+interface is also turned ON by default.
+
+The network interfaces can be referenced either by a ``struct net_if *``
+pointer or by a network interface index. The network interface can be
+resolved from index by calling ``net_if_get_by_index()`` and from interface
```
Comment (dbkinder) [inline doc/reference/networking/net_if.rst]:
```suggestion
resolved from its index by calling ``net_if_get_by_index()`` and from interface
```

File: doc/reference/networking/net_if.rst [hunk 3]
```diff
@@ -6,6 +6,50 @@ Network Interface
...
+pointer by calling ``net_if_get_by_iface()``.
+
+In order the network device to be connectable, then IP addresses can be set
```
Comment (dbkinder) [inline doc/reference/networking/net_if.rst]: How about:
```
The IP address for network devices must be set for them to be connectable.
In a typical dynamic network environment, IP addresses are set automatically
by DHCPv4, for example. If needed though, the application can set a device's
IP address manually.  See the API documentation below for functions such as 
``net_if_ipv4_addr_add()`` that do that.  
```

File: doc/reference/networking/net_if.rst [hunk 4]
```diff
@@ -6,6 +6,50 @@ Network Interface
...
+by various functions like ``net_if_ipv4_addr_add()``. See API
+reference documentation below for functions that are available for that.
+Typically in dynamic network environment the IP addresses are set automatically
+by DHCPv4 for example. If really needed the application can set the IP address
+itself.
+
+The ``net_if_get_default()`` returns a *default* network interface. What
+this default interface means can be configured via options like
+:option:`CONFIG_NET_DEFAULT_IF_FIRST` and
+:option:`CONFIG_NET_DEFAULT_IF_ETHERNET`. See Kconfig what options
```
Comment (dbkinder) [inline doc/reference/networking/net_if.rst]: Which Kconfig file should they look at?

**Review by dbkinder — DISMISSED** [review]

File: doc/reference/networking/net_if.rst (hunk 3 above)
Comment (jukkar) [reply doc/reference/networking/net_if.rst]: Yes, better. Fixed.

File: doc/reference/networking/net_if.rst (hunk 4 above)
Comment (jukkar) [reply doc/reference/networking/net_if.rst]: I will add the ref to the file.

**Comment by jukkar** [comment]
Updated according to comments.

@nashif sanity will complain about complex macros and while loops. This is false positive in this case.

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
Looks good, thanks!

**Pull request merged**